it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,364 @@
+    prepareCreateDatabaseMock();
+    prepareUpdateDatabaseMock();
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTableShouldThrowExceptionWhenSpannerCreateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareCreateInstanceMock();
+    when(spanner.getDatabaseAdminClient().createDatabase(any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    prepareUpdateDatabaseMock();
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTableShouldThrowExceptionWhenSpannerUpdateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareCreateInstanceMock();
+    prepareCreateDatabaseMock();
+    when(spanner.getDatabaseAdminClient().updateDatabaseDdl(any(), any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTableShouldWorkWhenSpannerDoesntThrowAnyError()
+      throws ExecutionException, InterruptedException {
+    //   arrange
+    prepareCreateInstanceMock();
+    prepareCreateDatabaseMock();
+    prepareUpdateDatabaseMock();
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act
+    testManager.createTable(statement);
+
+    // assert
+    verify(spanner.getDatabaseAdminClient(), times(2))
+        .updateDatabaseDdl(
+            instanceIdCaptor.capture(),
+            databaseIdCaptor.capture(),
+            statementCaptor.capture(),
+            any());
+
+    String actualInstanceId = instanceIdCaptor.getValue();
+    String actualDatabaseId = databaseIdCaptor.getValue();
+    Iterable<String> actualStatement = statementCaptor.getValue();
+
+    assertThat(actualInstanceId).startsWith(TEST_ID);
```

Comment:
As mentioned before, could we use a regex to match the instance id? That'll be clearer and more complete than `startsWith` and `isNotEqual`.

https://truth.dev/api/0.41/com/google/common/truth/StringSubject.html#matches-java.lang.String-